src/combinatorics/simple_graph/basic.lean

```diff
@@ -363,6 +364,35 @@ begin
   exact ⟨v, by simp, finset.subset_univ _⟩,
 end
 
+lemma card_common_neighbors_le_degree (v w : V) :
+  fintype.card (G.common_neighbors v w) ≤ G.degree v :=
+begin
+  rw [←card_neighbor_set_eq_degree],
+  exact set.card_le_of_subset (set.inter_subset_left _ _),
+end
+
+lemma card_common_neighbors_lt_card_verts (v w : V) :
+  fintype.card (G.common_neighbors v w) < fintype.card V :=
+nat.lt_of_le_of_lt (G.card_common_neighbors_le_degree _ _) (G.degree_lt_card_verts v)
+
+lemma card_common_neighbors_lt_degree_of_adj (v w : V) (h : G.adj v w) :
+  fintype.card (G.common_neighbors v w) < G.degree v :=
```

Comment:
It might be nice to add docstrings to this and `card_common_neighbors_le_degree` saying that if the condition `G.adj v w` fails, then the `le` bound you did above is the best we can do in general